### PR TITLE
[WIP] community/py-jsonschema: add check by import

### PR DESCRIPTION
**Pull request by ad-m**
See #7885 for consequences of not having a check.

### CODE REVIEW

**Comment by ad-m** [comment]
Superset #7885

**Comment by ad-m** [comment]
I do not see the cause of the error. Should I split the pull request and report aports to merge in the correct order?

File: community/py-jsonschema/APKBUILD [hunk 1]
```diff
@@ -8,7 +8,8 @@ pkgdesc="An implementation of JSON Schema validation for Python"
 url="https://github.com/Julian/jsonschema"
 arch="noarch"
 license="MIT"
-makedepends="python3-dev python2-dev py-setuptools"
+depends=""
```
Comment (Cogitri) [inline community/py-jsonschema/APKBUILD]: Why add an empty depends here?

File: community/py-jsonschema/APKBUILD [hunk 2]
```diff
@@ -8,7 +8,8 @@ pkgdesc="An implementation of JSON Schema validation for Python"
...
+makedepends="python3-dev python2-dev py2-functools32 py-pyrsistent py-attrs py-setuptools"
```
Comment (Cogitri) [inline community/py-jsonschema/APKBUILD]: Shouldn't the added makedepends be checkdepends?

File: testing/py-attrs/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+makedepends="python2-dev python3-dev py-setuptools"
+subpackages="py2-${pkgname#py-}:_py2 py3-${pkgname#py-}:_py3"
+source="https://files.pythonhosted.org/packages/source/${_pkgname:0:1}/$_pkgname/$_pkgname-$pkgver.tar.gz"
+builddir="$srcdir"/$_pkgname-$pkgver
+
+build() {
+	cd "$builddir"
+	python2 setup.py build || return 1
+	python3 setup.py build || return 1
+}
```
Comment (Cogitri) [inline testing/py-attrs/APKBUILD]: No need to cd into builddir, abuild does that automatically.
No need to return 1

File: testing/py-attrs/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+check() {
+	cd "$builddir"
```
Comment (Cogitri) [inline testing/py-attrs/APKBUILD]: No need to cd

File: testing/py-pyrsistent/APKBUILD [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+pkgname=py-pyrsistent
+_pkgname=${pkgname#py-}
+pkgver=0.15.2
+pkgrel=0
+pkgdesc="Pyrsistent is a number of persistent collections. Persistent in the sense that they are immutable."
+url="https://github.com/tobgu/pyrsistent/"
+arch="noarch"
+license="MIT"
+depends="py-six"
+makedepends="$depends_dev python2-dev python3-dev py-setuptools"
```
Comment (Cogitri) [inline testing/py-pyrsistent/APKBUILD]: There's no depends_dev hin this APKBUILD

File: testing/py-pyrsistent/APKBUILD [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+subpackages="py3-$_pkgname:_py3:all py2-$_pkgname:_py2:all"
+source="$pkgname-$pkgver.tar.gz::https://github.com/tobgu/pyrsistent/archive/v$pkgver.tar.gz"
+builddir="$srcdir/$_pkgname-$pkgver"
+
+build() {
+	cd "$builddir"
```
Comment (Cogitri) [inline testing/py-pyrsistent/APKBUILD]: No need to cd

File: testing/py-pyrsistent/APKBUILD [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+	python2 setup.py build
+	python3 setup.py build
+}
+
+check() {
+	cd "$builddir"
```
Comment (Cogitri) [inline testing/py-pyrsistent/APKBUILD]: No need to cd

File: testing/py-pyrsistent/APKBUILD [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+	python2 setup.py check
+	python3 setup.py check
+	python2 -c 'import pyrsistent';
+	python3 -c 'import pyrsistent';
+}
```
Comment (Cogitri) [inline testing/py-pyrsistent/APKBUILD]: Why import it aftet checking?

File: community/py-jsonschema/APKBUILD (hunk 2 above)
Comment (Ikke) [reply community/py-jsonschema/APKBUILD]: `py-attrs` and `py-jsonschema` have been added already in #7115 (as depends, not makedepends), so they can be skipped here.

**Pull request closed without merging**